test/e2e/__init__.py

```diff
@@ -30,4 +33,68 @@ def load_autoscaling_resource(resource_name: str, additional_replacements: Dict[
     """ Overrides the default `load_resource_file` to access the specific resources
     directory for the current service.
     """
-    return load_resource_file(resource_directory, resource_name, additional_replacements=additional_replacements)
\ No newline at end of file
+    return load_resource_file(resource_directory, resource_name, additional_replacements=additional_replacements)
+
+# TODO: Use this method to create resources instead
+def create_autoscaling_resource(
+    resource_plural, resource_name, spec_file, replacements, namespace="default"
+):
+    """
+    Wrapper around k8s.load_and_create_resource to create a SageMaker resource
+    """
+
+    reference, spec, resource = k8s.load_and_create_resource(
+        resource_directory,
+        CRD_GROUP,
+        CRD_VERSION,
+        resource_plural,
+        resource_name,
+        spec_file,
+        replacements,
+        namespace,
+    )
+
+    return reference, spec, resource
+
+
+def wait_for_status(
+    expected_status: str,
+    wait_periods: int,
+    period_length: int,
+    get_status_method,
+    *method_args,
+):
+    actual_status = None
+    for _ in range(wait_periods):
+        time.sleep(period_length)
+        actual_status = get_status_method(*method_args)
+        if actual_status == expected_status:
+            break
+    else:
+        logging.error(
+            f"Wait for status: {expected_status} timed out. Actual status: {actual_status}"
+        )
+
+    return actual_status
+
+
+def get_endpoint_sagemaker_status(sagemaker_client, endpoint_name):
+    response = sagemaker_client.describe_endpoint(EndpointName=endpoint_name)
+    return response["EndpointStatus"]
+
+
+def wait_sagemaker_endpoint_status(
```

Comment:
nit: SageMaker specific helper methods can be moved into their own utilities script, perhaps? It's a little confusing seeing them in `__init__.py`

Reply:
Fair point.